servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAliveManager.java

```diff
@@ -317,9 +323,12 @@ private void doCloseAsyncGracefully0(final Runnable whenInitiated) {
         channel.write(goAwayFrame);
         channel.writeAndFlush(new DefaultHttp2PingFrame(GRACEFUL_CLOSE_PING_CONTENT)).addListener(future -> {
             assert channel.eventLoop().inEventLoop();
-            // If gracefulCloseState is not GRACEFUL_CLOSE_START that means we have already received the PING(ACK) and
-            // there is no need to apply the timeout.
-            if (future.isSuccess() && gracefulCloseState == State.GRACEFUL_CLOSE_START) {
+            if (!future.isSuccess()) {
+                LOGGER.debug("{} Failed writing the first GO_AWAY and PING frames", channel, future.cause());
+                close0(future.cause());
```

Comment:
Consider mentioning in the log that the cannel is now going to be proactively closed.